freedomofpress/securedrop
# Build Rust redwood wheel during packaging process

## Status

Ready for review

## Description of Changes

* Properly exclude custom_logo.png from package and fix test (mostly unrelated, but discovered while working on this so I'm throwing it in...)
* Drop dh-virtualenv from packaging process 
* Build redwood wheel without maturin
* Build Rust redwood wheel during packaging process

Fixes https://github.com/freedomofpress/securedrop/issues/6538.
Fixes https://github.com/freedomofpress/securedrop/issues/6817.

## Testing

How should the reviewer test this PR?

Largely focusing on the dh-virtualenv change.

* [x] `cp securedrop/static/i/logo.png securedrop/static/i/custom_logo.png` (this should still be excluded from the build)
* [x] Check out this PR and run `make build-debs`
* [x] Download `https://apt.freedom.press/pool/main/s/securedrop/securedrop-app-code_2.6.0%2Bfocal_amd64.deb`, then run `debdiff` between the two files, you should see https://gist.github.com/legoktm/170f170e01437ba49671306df4425912
  * One intentional change is that `/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/securedrop_app_code-2.6.0-py3.8.egg-info/` is gone, we're no longer installing an empty securedrop-app-code Python package because dh_virtualenv mandates it.

For the Rust stuff:
* [x] deb-tests CI passes, which means the redwood.so is included in the package
* [x] staging CI passes, which means the redwood wheel is installed and usable for generating a key pair
* [x] review the `make build-debs` log from earlier, verify there's no garbage output (escape codes or progress bars) caused by Rust/cargo stuff

Verify Rust reproducibility:
* [x] Copy the securedrop-app-code.deb from your first `make build-debs` to something like `first.deb`
* [x] Re-run `make build-debs`, save the securedrop-app-code deb as something like `second.deb`
* [x] For each deb, extract the files. `ar x first.deb`, then `tar xvf data.tar.xz`
* [x] For each extracted deb, run `sha256sum opt/venvs/securedrop-app-code/lib/python3.8/site-packages/redwood/redwood.cpython-38-x86_64-linux-gnu.so` - the checksum should be `152d90961bf1cf44fcbc5c0275e31c68ca6a16f1b0c42895fe9272a53e2f1685` for both.

## Deployment

Any special considerations for deployment? Just the packaging changes.

## Checklist
